### PR TITLE
Rename set functions and deprecate

### PR DESCRIPTION
**Pull request by arifBurakDemiray**

### CODE REVIEW

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
Rework changelog closer to:
"Deprecated the init time configuration of 'setViewTracking(enable)'. Introduced replacement 'enableAutomaticViewTracking()'"

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+23.09.0
+* Deprecated the init time configuration of 'setEventsBufferSize(eventsBufferSize)'. Introduced replacement 'setEventQueueSizeToSend(eventsQueueSize)
+* Deprecated the init time configuration of 'setSendUpdateEachSeconds(sendUpdateEachSeconds)'. Introduced replacement 'setUpdateSessionTimerDelay(delay)
+
```
Comment (turtledreams) [inline CHANGELOG.md:4]: I think this is nice 👍  ( ' ) missing at the end for both line though. 
What do you guys think about showing the parameters ? mb unnecessary?

File: app-java/src/main/java/ly/count/java/demo/Sample.java [hunk 2]
```diff
@@ -132,7 +132,7 @@ public void LogHappened(String logMessage, Config.LoggingLevel logLevel) {
                     //System.out.println("[" + logLevel + "] " + logMessage);
                 }
             })
-            .setEventsBufferSize(1)//this should be only used for testing
+            .setEventQueueSizeToSend(1)//this should be only used for testing
```
Comment (turtledreams) [inline app-java/src/main/java/ly/count/java/demo/Sample.java:135]: this comment still holds?

File: CHANGELOG.md (hunk 1 above)
Comment (ArtursKadikis) [reply CHANGELOG.md:4]: yup, the ( ' ) are missing.
I feel like showing function parameters is a nice touch and probably for the best

File: sdk-java/src/main/java/ly/count/sdk/java/Config.java [hunk 3]
```diff
@@ -865,13 +866,34 @@ public Config setSendUpdateEachSeconds(int sendUpdateEachSeconds) {
...
+     *
+     * Update request is also sent when number of unsent events reached {@link #setEventQueueSizeToSend(int)}.
+     *
+     * @param delay max time interval between two update requests, set to 0 to disable update requests based on time.
+     * @return {@code this} instance for method chaining
+     */
+    public Config setUpdateSessionTimerDelay(int delay) {
+        if (sendUpdateEachSeconds < 0) {
+            if (configLog != null) {
+                configLog.e("[ConfigCore] sendUpdateEachSeconds cannot be negative");
```
Comment (turtledreams) [inline sdk-java/src/main/java/ly/count/sdk/java/Config.java:881]: this seems to be the old log, what do you think?

File: app-java/src/main/java/ly/count/java/demo/Sample.java (hunk 2 above)
Comment (ArtursKadikis) [reply app-java/src/main/java/ly/count/java/demo/Sample.java:135]: I guess the following would be a better message:
'setting queue size to "1" should only be done for testing, unless you have a really good reason to do it'

File: sdk-java/src/main/java/ly/count/sdk/java/Config.java [hunk 4]
```diff
@@ -884,13 +906,32 @@ public Config setEventsBufferSize(int eventsBufferSize) {
...
+     *
+     * Update request is also sent when last update request was sent more than {@link #setUpdateSessionTimerDelay(int)} seconds ago.
+     *
+     * @param eventsQueueSize max number of events between two update requests, set to 0 to disable update requests based on events.
+     * @return {@code this} instance for method chaining
+     */
+    public Config setEventQueueSizeToSend(int eventsQueueSize) {
+        if (eventsBufferSize < 0) {
+            if (configLog != null) {
+                configLog.e("[ConfigCore] eventsBufferSize cannot be negative");
```
Comment (turtledreams) [inline sdk-java/src/main/java/ly/count/sdk/java/Config.java:920]: similar to above?

File: CHANGELOG.md (hunk 1 above)
Comment (arifBurakDemiray) [reply CHANGELOG.md:4]: got it

File: sdk-java/src/main/java/ly/count/sdk/java/Config.java (hunk 3 above)
Comment (arifBurakDemiray) [reply sdk-java/src/main/java/ly/count/sdk/java/Config.java:881]: Should I change all ConfigCore occurences? @ArtursKadikis

Comment (ArtursKadikis) [reply sdk-java/src/main/java/ly/count/sdk/java/Config.java:881]: yup, pls do

Comment (arifBurakDemiray) [reply sdk-java/src/main/java/ly/count/sdk/java/Config.java]: done

File: sdk-java/src/main/java/ly/count/sdk/java/Config.java (hunk 4 above)
Comment (arifBurakDemiray) [reply sdk-java/src/main/java/ly/count/sdk/java/Config.java]: got it

**Review by ArtursKadikis — APPROVED** [review]

File: sdk-java/src/main/java/ly/count/sdk/java/Config.java (hunk 4 above)
Comment (yelimot) [reply sdk-java/src/main/java/ly/count/sdk/java/Config.java]: nice work 👍

**Pull request merged**